### DISCUSSION TITLE
A single PR vs several PRs when implementing new models

### DISCUSSION THREAD

**Discussion by lbluque** [materialyzeai/matgl#86] [Q&A] [answered]
Hi matgl team :wave: 

I have a question on the development and contributing workflow when implementing a new model that requires additional layers and functionality and possibly edits to existing functionality.

1) Would you prefer a single large PR with all changes to only review code changes once?  
2) Have multiple but specific PRs to keep the reviews simpler? For example, a PR when editing an existing layer, or a PR with the implementation of a new general-use case layer?

I think the latter is better because although it involves multiple reviews on your end (thanks!), it prevents potentially duplicated code and long-running feature development to diverge too far from the core design. I am interested to hear your suggestions on how to best move forward.

**Comment by shyuep** [comment, marked as answer]
Many small PRs are always preferred and generally the recommended way to manage open-source projects.

**Reply by lbluque** [reply]
Thanks @shyuep  will try to proceed by making small PRs

**Comment by shyuep** [comment]
I would add that if there are changes that are backward incompatible, e.g., it will break one of the pre-trained models, this should be made known in the PR. But the organization is such that changes to the internals shouldn't create too many problems.
